proposals/paddle-operator-proposal.md

```diff
@@ -0,0 +1,236 @@
+  resourceVersion: "9513616"
+  selfLink: /api/v1/namespaces/paddle-system/services/wide-ande-deep-service-ps-0
+  uid: e274db1e-ee7f-4b6d-bc0c-034c32f4b7a1
+spec:
+  clusterIP: None
+  ports:
+  - port: 2379
+    protocol: TCP
+    targetPort: 2379
+  selector:
+    paddle-res-name: wide-ande-deep-service-ps-0
+  sessionAffinity: None
+  type: ClusterIP
+```
+
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: wide-ande-deep-ps-0
+  namespace: paddle-system
+  ownerReferences:
+  - apiVersion: batch.paddlepaddle.org/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: PaddleJob
+    name: wide-ande-deep
+    uid: f206587f-5dee-46f5-9399-e835bde02487
+  resourceVersion: "9506900"
+  selfLink: /api/v1/namespaces/paddle-system/pods/wide-ande-deep-ps-0
+  uid: 36b27c8f-9712-474b-b21b-dd6b54aaef29
+spec:
+  containers:
+  - env:
+    - name: POD_IP
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: status.podIP
+    - name: PADDLE_TRAINER_ID
+      value: "0"
+    - name: TRAINING_ROLE
+      value: PSERVER
+    - name: PADDLE_TRAINING_ROLE
+      value: PSERVER
+    envFrom:
+    - configMapRef:
+        name: wide-ande-deep
+    image: registry.baidubce.com/paddle-operator/demo-wide-and-deep:v1
+```
+
+### Resulting Worker
+```yaml
+apiVersion: v1
+kind: Pod
+metadata:
+  name: wide-ande-deep-worker-0
+  namespace: paddle-system
+  ownerReferences:
+  - apiVersion: batch.paddlepaddle.org/v1
+    blockOwnerDeletion: true
+    controller: true
+    kind: PaddleJob
+    name: wide-ande-deep
+    uid: f206587f-5dee-46f5-9399-e835bde02487
+  resourceVersion: "9507629"
+  selfLink: /api/v1/namespaces/paddle-system/pods/wide-ande-deep-worker-0
+  uid: e8534fe6-7c2e-4849-9a99-ffdcd5df76bb
+spec:
+  containers:
+  - env:
+    - name: POD_IP
+      valueFrom:
+        fieldRef:
+          apiVersion: v1
+          fieldPath: status.podIP
+    - name: PADDLE_TRAINER_ID
+      value: "0"
+    - name: TRAINING_ROLE
+      value: TRAINER
+    - name: PADDLE_TRAINING_ROLE
+      value: TRAINER
+    envFrom:
+    - configMapRef:
+        name: wide-ande-deep
+    image: registry.baidubce.com/paddle-operator/demo-wide-and-deep:v1
+```
+
+The worker spec generates a pod. Currently worker will communicate to the master through the master's service name, we'll use a service registry for service discovery.
+
+## Design
+
+Here are some original design docs for paddle-perator on Kubernetes.
+ 
+* Paddle Operator Architecture on Kubernetes, please check out [design-arch](https://github.com/PaddleFlow/paddle-operator/blob/main/docs/design-arch.md)  
+* Paddle training job instance fault tolerant, please check out [design-fault-tolerant](https://github.com/PaddleFlow/paddle-operator/blob/main/docs/design-fault-tolerant.md)
+* Co-scheduling training job to prevent job instances from resource deadlock, please check out [design-coschedule](https://github.com/PaddleFlow/paddle-operator/blob/main/docs/design-coschedule.md)
```

Comment:
All link expired

Reply:
Since we've just merged the new refactoring, I'll fix it soon.